ext/js/dom/text-source-range.js

```diff
@@ -95,6 +95,18 @@ export class TextSourceRange {
         return this._imposterSourceElement;
     }
 
+    /**
+     * Determines whether the imposter source element is an input or textarea element.
+     * @return {boolean} `true` if the imposter source element is an input or textarea element, `false` otherwise.
+     */
+     isImposterInputOrTextArea() {
+        if (this._imposterSourceElement) {
+            const type = this._imposterSourceElement.nodeName.toUpperCase();
+            return type === 'INPUT' || type === 'TEXTAREA';
+        }
+        return false;
+     }
+
```

Comment:
From the code, looks like `imposterSourceElement` can only be `input` or `textarea` if not null. This may be redundant?